xdsl/ir/core.py

```diff
@@ -1087,6 +1087,7 @@ def detach(self):
         if self.parent is None:
             raise Exception("Cannot detach a toplevel operation.")
         self.parent.detach_op(self)
+        return self
```

Comment:
Why do you need this? You have the object when calling `detach()` on it, right? If you just need it like this in this specific lowering, I would prefer wrapping this into a local lambda rather than changing core. Otherwise, this should probably be PRed (and tested) individually, IMO.